cairo/ethereum/cancun/fork.cairo

```diff
@@ -540,3 +566,140 @@ func process_storage_keys{
+    let sender_account = get_account{state=state}(sender_address);
+    let transaction_type = TransactionImpl._get_transaction_type(tx);
+    let is_not_blob_or_fee_transaction = (TransactionType.BLOB - transaction_type) * (
+        TransactionType.FEE_MARKET - transaction_type
+    );
+    // Case where transaction is blob or fee transaction
+    if (is_not_blob_or_fee_transaction == FALSE) {
+        let max_fee_per_gas = TransactionImpl.get_max_fee_per_gas(tx);
+        let max_priority_fee_per_gas = TransactionImpl.get_max_priority_fee_per_gas(tx);
+        let max_fee_per_gas_valid = is_le(base_fee_per_gas.value, max_fee_per_gas.value);
+        let max_priority_fee_per_gas_valid = is_le(
+            max_priority_fee_per_gas.value, max_fee_per_gas.value
+        );
+        let is_coherent_fee = max_fee_per_gas_valid * max_priority_fee_per_gas_valid;
+        with_attr error_message("InvalidBlock") {
+            assert is_coherent_fee = 1;
+        }
+        let priority_fee_per_gas = min(
+            max_priority_fee_per_gas.value, max_fee_per_gas.value - base_fee_per_gas.value
+        );
+        tempvar effective_gas_price = Uint(base_fee_per_gas.value + priority_fee_per_gas);
+        tempvar max_gas_fee = Uint(gas.value * max_fee_per_gas.value);
+    } else {
+        let gas_price = TransactionImpl.get_gas_price(tx);
+        let gas_price_valid = is_le(base_fee_per_gas.value, gas_price.value);
+        with_attr error_message("InvalidBlock") {
+            assert gas_price_valid = 1;
+        }
+        tempvar effective_gas_price = gas_price;
+        tempvar max_gas_fee = Uint(gas.value * gas_price.value);
+    }
+    let effective_gas_price = effective_gas_price;
+    let max_gas_fee = max_gas_fee;
+    if (transaction_type == TransactionType.BLOB) {
+        let len = tx.value.blob_transaction.value.blob_versioned_hashes.value.len;
+        with_attr error_message("InvalidBlock") {
+            assert_not_zero(len);
+        }
+        // Check that each versioned hash has the correct version, i.e. that blob_versioned_hash[0:1] == VERSIONED_HASH_VERSION_KZG
+        // for each versioned_hash in blob_versioned_hashes
+        tempvar index = len;
+        tempvar range_check_ptr = range_check_ptr;
+
+        loop:
+        let index = [ap - 2];
+        let range_check_ptr = [ap - 1];
+        let versioned_hash = tx.value.blob_transaction.value.blob_versioned_hashes.value.data[
+            index
+        ];
+        // Since versioned_hash are hash32 which are little endian, we need to check that the least significant byte is 0x01
+        let (first_byte, _) = divmod(versioned_hash.value.low, 2 ** 128);
+        with_attr error_message("InvalidBlock") {
+            assert first_byte = VERSIONED_HASH_VERSION_KZG;
+        }
+        tempvar index = index - 1;
+        tempvar range_check_ptr = range_check_ptr;
+
+        jmp loop if index != 0;
+        static_assert range_check_ptr == [ap - 1];
```

Comment:
do the static_assert before the jump to make sure that at the next iterations values are what you expect them to be